src/README.md

```diff
@@ -0,0 +1,225 @@
+# YNAB Toolkit Development
+This folder holds all the source code for the YNAB Toolkit.
+
+The folder structure is as such:
+
+```
+|-core
+|---listeners
+|-features
+|---accounts
+|---budget
+|---general
+|---reports
+|---toolkitReports
+|-helpers
+```
+
+`core/`: Contains all the core functionality for the Toolkit such as the observers
+and the base `Feature` class.
+
+`features/`: Contains sub-directories for each section of the YNAB application
+and house the source code for each individual feature. This is where most
+development will take place.
+
+`helpers/`: Contains any helpers shared code for the Toolkit such as looking up
+Ember views or normalizing currency values.
+
+## Writing Your First Feature
+
+It is extremely easy to get started with your first feature. In order to do so,
+follow these steps:
+
+1. Determine where your feature belongs in YNAB (accounts/budget/general)
+2. Create a sub-directory in the proper sub-directory e.g. `account/`, `budget/`, `general/`
+3. Create an `index.js` file which has the following:
+  <!-- spacing is intentionally weird here because of markdown -->
+  ```javascript
+  import { Feature } from 'toolkit/extension/features/feature';
+
+  export class MyCoolFeature extends Feature {
+    shouldInvoke() {
+      return true;
+    }
+
+    invoke() {
+      console.log('MyCoolFeature is working!');
+    }
+  }
+  ```
+4. Create a `settings.js` file which has the following:
+  <!-- spacing is intentionally weird here because of markdown -->
+  ```javascript
+  module.exports = {
+    name: 'MyCoolFeature',
+    type: 'checkbox',
+    default: false,
+    section: 'budget',
+    title: 'My Cool Feature!',
+    description: 'This is my brand new feature.'
+  };
+
+  ```
+
+5. Run `./build` to build the extension for all the browswers.
+6. *chrome:* go to `chrome://extensions` and turn on "Developer mode". Then "Load
+unpacked extension". Select `/output/chrome/` and it will load into chrome.
+7. Reload YNAB!
+
+In order to help you develop cool features, we've created a few API functions
+that you get for free when extending `Feature`.
+
+### The `Feature` Class
+
+#### The following functions can be declared inside your Feature Class.
+
+#### `constructor()`
+**optional function, not required to be declared**
+
+Your feature's constructor is invoked as soon as the Toolkit is injected onto
+the page. You should not attempt a DOM manipulation or access to Ember/YNAB
+as it is not guaranteed or likely to be ready when your constructor is invoked.
+The job of the base `Feature` constructor is to simply fetch the user settings
```

Comment:
Initialise rather than fetch user settings I reckon. We should also note that usually it's not required.